src/PIL/EpsImagePlugin.py

```diff
@@ -324,32 +325,50 @@ def _read_comment(s: str) -> bool:
                     int(value) for value in image_data_values[:4]
                 )
 
-                if bit_depth == 1:
-                    self._mode = "1"
-                elif bit_depth == 8:
-                    try:
-                        self._mode = self.mode_map[mode_id]
-                    except ValueError:
-                        break
-                else:
-                    break
+                if not imagedata_size:
+                    imagedata_size = columns, rows
 
-                self._size = columns, rows
-                return
+                    if bit_depth == 1:
+                        self._mode = "1"
+                    elif bit_depth == 8:
+                        try:
+                            self._mode = self.mode_map[mode_id]
+                        except ValueError:
+                            pass
             elif bytes_mv[:5] == b"%%EOF":
                 break
             elif trailer_reached and reading_trailer_comments:
                 # Load EPS trailer
                 s = str(bytes_mv[:bytes_read], "latin-1")
-                _read_comment(s)
+                read_comment(s)
             elif bytes_mv[:9] == b"%%Trailer":
                 trailer_reached = True
             bytes_read = 0
 
-        if not self._size:
+        # A "BoundingBox" is always required,
+        # even if an "ImageData" descriptor size exists.
+        if not bounding_box:
             msg = "cannot determine EPS bounding box"
             raise OSError(msg)
 
+        # An "ImageData" size takes precedence over the "BoundingBox".
+        if imagedata_size:
+            self._size = imagedata_size
+        else:
+            self._size = (
+                bounding_box[2] - bounding_box[0],
+                bounding_box[3] - bounding_box[1],
+            )
+
+        self.tile = [
+            ImageFile._Tile(
+                codec_name="eps",
+                extents=(0, 0) + self._size,
+                offset=offset,
+                args=(length, bounding_box),
+            )
```

Comment:
I realise mentioning the names for the NamedTuple would just be your personal preference, but ideally Pillow's code would be somewhat consistent, rather than just a combination of the styles of whoever worked on a given plugin last.

See, for example, JpegImagePlugin, where the tile is created without names.

https://github.com/python-pillow/Pillow/blob/0ec1153a627a46b978022c68c2adce89ff81f40d/src/PIL/JpegImagePlugin.py#L829-L831